### PR TITLE
Add disabled button guidence

### PR DESCRIPTION
**Pull request by notlee**
https://github.com/Financial-Times/o-buttons/issues/256

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,7 +117,12 @@ For a pagination style wrap your buttons in `.o-buttons-pagination`. In this exa
 ```
 
 ### Disabled
-`o-buttons` styles buttons with the `disabled` attribute to show they are inactive. In some cases it is preferred to visually hide the button until it is active. For this case add the class `o-buttons--hide-disabled`.
+
+Styles for disabled states are included in `o-buttons` however the use of disabled buttons is not recommended. Firstly disabled buttons pose accessibility issues including being difficult to see. Although disabled buttons with low contrast are allowed according to [Web Content Accessibility Guidelines](https://www.w3.org/TR/WCAG21/#abstract), blind or visually impaired users may miss vital context. Secondly disabled buttons may be frustrating for the user if they are unsure why the button is disabled.
```
Comment (JakeChampion) [inline README.md:121]:
```suggestion
Styles for disabled states are included in `o-buttons` however the use of disabled buttons is not recommended. Firstly, disabled buttons pose accessibility issues including being difficult to see. Although disabled buttons with low contrast are allowed according to [Web Content Accessibility Guidelines](https://www.w3.org/TR/WCAG21/#abstract), blind or visually impaired users may miss vital context. Secondly, disabled buttons may be frustrating for the user if they are unsure why the button is disabled.
```

**Comment by origamiserviceuser** [comment]
o-buttons bundle size difference from 6.0.15 to 24335e8ec43967ccb60c4abbe0195529c008dc80
No bundle size differences found.

**Comment by origamiserviceuser** [comment]
o-buttons bundle size difference from 6.0.15 to 7a16bf4b865edd016ed3db7082b82ac4f851a98f
No bundle size differences found.

File: README.md [hunk 2]
```diff
@@ -117,7 +117,12 @@ For a pagination style wrap your buttons in `.o-buttons-pagination`. In this exa
 ```
 
 ### Disabled
-`o-buttons` styles buttons with the `disabled` attribute to show they are inactive. In some cases it is preferred to visually hide the button until it is active. For this case add the class `o-buttons--hide-disabled`.
+
+Styles for disabled states are included in `o-buttons` however the use of disabled buttons is not recommended. Firstly, disabled buttons pose accessibility issues including being difficult to see. Although disabled buttons with low contrast are allowed according to [Web Content Accessibility Guidelines](https://www.w3.org/TR/WCAG21/#abstract), blind or visually impaired users may miss vital context. Secondly, disabled buttons may be frustrating for the user if they are unsure why the button is disabled.
+
+Hampus Sethfors explains the issues with disabled buttons in more detail in an [axesslab.com article](https://axesslab.com/disabled-buttons-suck/) and suggests alternative design approaches for common uses of disabled buttons.
```
Comment (nickcolley) [inline README.md:123]: Suggest to move to references

File: README.md [hunk 3]
```diff
@@ -117,7 +117,12 @@ For a pagination style wrap your buttons in `.o-buttons-pagination`. In this exa
 ```
 
 ### Disabled
-`o-buttons` styles buttons with the `disabled` attribute to show they are inactive. In some cases it is preferred to visually hide the button until it is active. For this case add the class `o-buttons--hide-disabled`.
+
+Styles for disabled states are included in `o-buttons` however the use of disabled buttons is not recommended. Firstly, disabled buttons pose accessibility issues including being difficult to see. Although disabled buttons with low contrast are allowed according to [Web Content Accessibility Guidelines](https://www.w3.org/TR/WCAG21/#abstract), blind or visually impaired users may miss vital context. Secondly, disabled buttons may be frustrating for the user if they are unsure why the button is disabled.
```
Comment (nickcolley) [inline README.md:121]:
```suggestion
Disabled buttons have [poor contrast which makes them difficult to read](). Disabled buttons [do not give feedback to a user why they are disabled]().

Avoid them unless user research shows they improve your interface.
```

I would move some of the more academic nuances into a reference section at the bottom or in the commit message itself, as we want to keep the content as minimal as possible to what the reader needs to understand.

File: README.md [hunk 4]
```diff
@@ -117,7 +117,12 @@ For a pagination style wrap your buttons in `.o-buttons-pagination`. In this exa
...
+
+Hampus Sethfors explains the issues with disabled buttons in more detail in an [axesslab.com article](https://axesslab.com/disabled-buttons-suck/) and suggests alternative design approaches for common uses of disabled buttons.
+
+To use a disabled button regardless of these issues add the `disabled` attribute. To visually hide the disabled button until it is active add the class `o-buttons--hide-disabled`.
```
Comment (nickcolley) [inline README.md:125]:
```suggestion
Add the `disabled` attribute to make a button disabled. To visually hide the disabled button until it is active add the class `o-buttons--hide-disabled`.
```

We can trust the reader to make the judgement and simplify this sentence.

**Comment by origamiserviceuser** [comment]
o-buttons bundle size difference from 6.0.15 to 2efd89c820fb1a0520445497aa859f9c9fa8471e
No bundle size differences found.

**Comment by notlee** [comment]
thanks @nickcolley, I like the idea of a references section to keep the main content minimal

**Comment by origamiserviceuser** [comment]
o-buttons bundle size difference from 6.0.15 to 734a221a3c6cf5c7eb767e713bd444e0c00be941
No bundle size differences found.

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v6.0.16 :tada:

**Pull request merged**